src/pkg/selectors/exchange.go

```diff
@@ -639,6 +641,15 @@ func (s ExchangeScope) setDefaults() {
 // Backup Details Filtering
 // ---------------------------------------------------------------------------
 
+func (s exchange) FolderRefToPath(ref string, deets *details.Details) (string, error) {
```

Comment:
I know yesterday we talked about the controller packages that call into selectors.Reduce call this package.  But in thinking about it further that approach is beginning to smell a little.  It requires the caller to fetch details before building the scopes that reduce the details, which feels off.

At the start of the Reduce call there's already a set of pre-processing being done to massage inputs into a more manageable state.  I think you could run a map over the scopes at that stage to transform all the folder-level scopes as needed.  Would only require a single pass over the details folder entries (o(n*m) for needing to compare each folder scope to a detail entry, but 'm' is pretty well guaranteed to be small), and then those changes would cascade onto the rest of the reduce.  Thoughts?

Reply:
Doing all this in a cleaner fashion sounds great. I know I picked this issue up, but I'm not sure I actually have the knowledge of the selectors package to figure out the part of the code to modify based off the description you gave. You'll either have to point me to where the code needs modified (hopefully it's more of lift'n'shift than anything) or, if you have bandwidth, you can finish this PR series/issue off

Reply:
> point me to where the code needs modified

Happily!  Around about [here](https://github.com/alcionai/corso/blob/main/src/pkg/selectors/scopes.go#L237) is where my mind is at.  For each set of scopes (excludes, filters, includes), we'd do a pre-pass like:
```go
for _, ent := range deets.Folders() {
  for _, s := range s.Excludes|Filers|Includes {
    scat := s.Category()
    if !scat.isLeaf() && scat != scat.RootCat() {
      // check folderRefToPath
      // if is ref to path, update (a copy of?) the scope to filter on the found path
    }
  }
}
```
Thoughts?

Reply:
hmmm thanks for the pointer.

Looking at the referenced code, a couple of thoughts:
* we still need to iterate through details 2x, once to collect the folders and once to collect the items. This would be true for any solution of this form and is something I failed to consider when writing the original proposal for it (🙄). Given that, maybe the strawman would be good enough for now? I really don't know
* embedding the resolution of folder ShortRefs in the reduce operation may make it difficult to determine what went wrong if nothing matches. For instance, it's hard to say if there was a typo in the ref as they're more or less random hex strings. On the other hand, given that this may not be available as CLI behavior it's probably less of a concern, though still a point about debuggability overall

Reply:
> we still need to iterate through details 2x

```go
func (dm DetailsModel) FoldersItems() ([]*detailsEntry, []*detailsEntry)
```

> difficult to determine what went wrong if nothing matches

I think that is an unfortunately present problem independent of data sourcing.  We can only do so much to protect against typos.  It's whether we're opening ourselves up to an inability to distinguish a typo from a legitimate bug that that would concern me more.  And, I think a more centralized and standardized handling is the best defense against that possibility.

> maybe the strawman would be good enough for now? I really don't know

Neither do I.  But I want to try something (that's hopefully a bit of fun) to help us find out.  We don't often have the breathing room to compare and contrast approaches.  Would you be interested in each of us taking on the problem in each way: the strawman for you, the pre-processer for me?  Have a quick code jam and see which PR we like better?

Reply:
>Would you be interested in each of us taking on the problem in each way: the strawman for you, the pre-processer for me?

Sure, seems like it could be interesting. I'll spin up a bit of code with the strawman

Reply:
![CODE JAM TIME](https://i.gifer.com/Ay64.gif)

Reply:
alright, there's now have #1349 to compare against